packages/happy-dom/src/nodes/document/Document.ts

```diff
@@ -723,7 +723,7 @@ export default class Document extends Node implements IDocument {
 		this._isFirstWriteAfterOpen = true;
 
 		for (const eventType of Object.keys(this._listeners)) {
-			const listeners = this._listeners[eventType];
+			const listeners = this._listeners[eventType].map((listener) => listener.fn);
```

Comment:
I think we can speed this up a bit by checking directly in the for loop instead as we can avoid the extra loop with the `map()`.

Reply:
Yes, you are right. 👍